### PR TITLE
fix(http): clear queues after hostrules change

### PR DESCRIPTION
**Pull request by viceice**
## Changes
clear http queues after hostrule change, as the concurrency config is probably no longer valid

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/http/queue.ts [hunk 1]
```diff
@@ -15,7 +18,10 @@ export function getQueue(url: string): PQueue | null {
     queue = null; // null represents "no queue", as opposed to undefined
     const concurrency = getRequestLimit(url);
     if (concurrency) {
+      logger.debug({ concurrency, host }, 'Using queue');
       queue = new PQueue({ concurrency });
+    } else {
+      logger.debug({ host }, 'No concurency limits');
```
Comment (rarkins) [inline lib/util/http/queue.ts:24]: Are these going to log a debug message for every single http request?

**Comment by rarkins** [comment]
Wouldn't it be better to clear queues from util/host-rules? e.g. any time a rule is added or the rules are cleared?

**Comment by viceice** [comment]
> Wouldn't it be better to clear queues from util/host-rules? e.g. any time a rule is added or the rules are cleared?

no, would cause circular references

File: lib/util/http/queue.ts (hunk 1 above)
Comment (viceice) [reply lib/util/http/queue.ts:24]: nope, once per host

**Review by rarkins — APPROVED** [review]

**Pull request merged**